system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <memory>
+#include <mutex>
+
+#include "../../src/system_metrics_collector/collector.hpp"
+#include "../../src/system_metrics_collector/periodic_measurement_node.hpp"
+#include "../../src/moving_average_statistics/types.hpp"
+
+
+static constexpr const std::chrono::milliseconds TEST_LENGTH =
+  std::chrono::milliseconds(250);
+static constexpr const std::chrono::milliseconds TEST_PERIOD =
+  std::chrono::milliseconds(50);
+
+/**
+ * Simple extension to test basic functionality
+ */
+class TestPeriodicMeasurementNode : public PeriodicMeasurementNode
+{
+public:
+  TestPeriodicMeasurementNode(
+    const std::string & name,
+    const std::chrono::milliseconds & measurement_period,
```

Comment:
```c++
const std::chrono::miliseconds measurement_period
```
sizeof(std::chrono::milliseconds) is 8, so passing by ref might cost more than passing by val
http://cpp.sh/2dndi

Reply:
I'm going to change all the constructors to take a val instead of ref. This follows the ROS2 timer definitions as well.